### PR TITLE
Ags/working workers

### PR DESCRIPTION
**Pull request by lalexgap**

### CODE REVIEW

File: packages/server-wallet/benchmarks/update-channel.benchmark.ts [hunk 1]
```diff
@@ -37,7 +37,9 @@ async function benchmark(): Promise<void> {
   await adminKnex.migrate.latest();
   const wallet = new Wallet(defaultConfig);
 
-  let channels = await setup(5);
+  // Warm up each worker thread.
+  // eslint-disable-next-line no-process-env
+  let channels = await setup(Number(process.env.AMOUNT_OF_WORKER_THREADS ?? 0));
```
Comment (lalexgap) [inline packages/server-wallet/benchmarks/update-channel.benchmark.ts:42]: Implement this

File: packages/server-wallet/e2e-test/e2e-utils.ts [hunk 2]
```diff
@@ -53,13 +51,10 @@ export const triggerPayments = async (
...
 export const startReceiverServer = (): ReceiverServer => {
-  const server = spawn('yarn', ['ts-node', './e2e-test/receiver/server'], {
+  const server = spawn('yarn', ['node', './lib/e2e-test/receiver/server'], {
     stdio: 'inherit',
-    env: {
-      AMOUNT_OF_WORKER_THREADS: '2',
-      // eslint-disable-next-line
-      ...process.env,
-    },
+    // eslint-disable-next-line no-process-env
```
Comment (lalexgap) [inline packages/server-wallet/e2e-test/e2e-utils.ts:56]: Do this

File: packages/server-wallet/e2e-test/e2e-utils.ts [hunk 3]
```diff
@@ -108,6 +103,10 @@ export async function seedTestChannels(
   knexPayer: Knex
 ): Promise<string[]> {
   const channelIds: string[] = [];
+
+  const payerSeeds = [];
```
Comment (lalexgap) [inline packages/server-wallet/e2e-test/e2e-utils.ts:107]: keep these

File: packages/server-wallet/e2e-test/stress.test.ts [hunk 4]
```diff
@@ -51,34 +51,16 @@ describe('Stress tests', () => {
     await SWReceiver.query(knexReceiver).insert(bob());
   });
 
-  it('runs the stress test with 100 channels and 1 payment call', async () => {
```
Comment (lalexgap) [inline packages/server-wallet/e2e-test/stress.test.ts:54]: delete this

File: packages/server-wallet/e2e-test/stress.test.ts [hunk 5]
```diff
@@ -51,34 +51,16 @@ describe('Stress tests', () => {
...
-
-    await triggerPayments(channelIds);
-
-    for (const channelId of channelIds) {
-      await await expectSupportedState(knexPayer, channelId, ChannelPayer, 5);
-      await expectSupportedState(knexReceiver, channelId, ChannelReceiver, 5);
-    }
-  });
-
   it('runs the stress test with 100 channels and 25 payment call', async () => {
```
Comment (lalexgap) [inline packages/server-wallet/e2e-test/stress.test.ts:54]: update numbers and update description

File: packages/server-wallet/package.json [hunk 6]
```diff
@@ -89,8 +89,8 @@
     "start:watch": "nodemon --watch 'src/**/*' -e ts,tsx --exec ts-node ./src/start-dev-server.ts | npx pino-pretty",
     "test": "yarn jest -c ./jest/jest.config.js --runInBand",
     "test:ci": "yarn test --bail",
-    "test:e2e": "SERVER_DB_NAME=payer yarn jest -c ./jest/jest.e2e.config.js",
-    "test:stress": "SERVER_DB_NAME=payer yarn jest -c ./jest/jest.stress.config.js --runInBand",
+    "test:e2e": "yarn tsc; SERVER_DB_NAME=payer yarn jest -c ./jest/jest.e2e.config.js",
+    "test:stress": "yarn tsc; SERVER_DB_NAME=payer yarn jest -c ./jest/jest.stress.config.js --runInBand",
```
Comment (lalexgap) [inline packages/server-wallet/package.json:93]: keep/ explain in commit message

File: packages/server-wallet/src/db/migrations/20200915140300_chain_service.ts [hunk 7]
```diff
@@ -8,7 +8,10 @@ const defaultFundingStrategy: FundingStrategy = 'Unknown';
 
 export async function up(knex: Knex): Promise<any> {
   await knex.schema.alterTable(channels, table => {
-    table.specificType(chainServiceRequests, 'text[]').notNullable();
+    table
+      .specificType(chainServiceRequests, 'text[]')
+      .notNullable()
+      .defaultTo('{}');
```
Comment (lalexgap) [inline packages/server-wallet/src/db/migrations/20200915140300_chain_service.ts:14]: @andrewgordstewart separate PR

File: packages/server-wallet/src/utilities/signatures.ts [hunk 8]
```diff
@@ -23,7 +23,12 @@ export async function fastSignState(
   return {state, signature};
 }
 
-export function fastRecoverAddress(state: State, signature: string, stateHash: string): string {
+export async function fastRecoverAddress(
```
Comment (lalexgap) [inline packages/server-wallet/src/utilities/signatures.ts:26]: Separate PR

File: packages/server-wallet/src/utilities/workers/loader.js [hunk 9]
```diff
@@ -1,21 +1,27 @@
+// const {threadId} = require('worker_threads');
 const path = require('path');
```
Comment (lalexgap) [inline packages/server-wallet/src/utilities/workers/loader.js:2]: Already done

File: packages/server-wallet/src/utilities/workers/worker.ts [hunk 10]
```diff
@@ -38,6 +39,8 @@ parentPort?.on('message', async (message: any) => {
         return parentPort?.postMessage(right(await wallet._updateChannel(message.args)));
       case 'PushMessage':
         return parentPort?.postMessage(right(await wallet._pushMessage(message.args)));
+      default:
```
Comment (lalexgap) [inline packages/server-wallet/src/utilities/workers/worker.ts:42]: Keep

File: packages/server-wallet/src/utilities/workers/worker.ts [hunk 11]
```diff
@@ -38,6 +39,8 @@ parentPort?.on('message', async (message: any) => {
...
+        return parentPort?.postMessage(left(new Error('Unknown message type')));
```
Comment (lalexgap) [inline packages/server-wallet/src/utilities/workers/worker.ts:43]: Add comment to see what happens to serialized error class

File: packages/server-wallet/src/wallet/store.ts [hunk 12]
```diff
@@ -379,11 +379,15 @@ async function getSigningWallet(
  * Validator functions
  */
 
-function validateSignatures(signedState: SignedStateWithHash): void {
+async function validateSignatures(signedState: SignedStateWithHash): Promise<void> {
   const {participants} = signedState;
 
-  signedState.signatures.map(sig => {
-    const signerAddress = fastRecoverAddress(signedState, sig.signature, signedState.stateHash);
+  signedState.signatures.map(async sig => {
```
Comment (lalexgap) [inline packages/server-wallet/src/wallet/store.ts:385]: separate PR

Comment (lalexgap) [reply packages/server-wallet/src/wallet/store.ts:385]: NM do it here

**Comment by andrewgordstewart** [comment]
Closing this PR, as it was created simply for bookkeeping purposes.

**Pull request closed without merging**